Build error with clang-12

The following code does not compile with Clang-12 (I am using macOS, but I think the problem is more general):

https://github.com/googleapis/google-cloud-cpp/blob/4eb5b7c257b549483730f7e8dadb8ee8583134e2/google/cloud/spanner/value.h#L423

The problem is when `T == bool`, and the iteration does not return references.

Also in one of the tests, I captured the error message this time:

```
/Users/coryan/cbt-develop/google/cloud/spanner/integration_tests/data_types_integration_test.cc:49:20: error: loop variable 'x' is always a copy because the range of type 'const std::__1::vector<bool, std::__1::allocator<bool> >' does not return a reference [-Werror,-Wrange-loop-analysis]
  for (auto const& x : data) {
                   ^
/Users/coryan/cbt-develop/google/cloud/spanner/integration_tests/data_types_integration_test.cc:94:17: note: in instantiation of function template specialization 'google::cloud::spanner::v1::(anonymous namespace)::WriteReadData<std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
  auto result = WriteReadData(*client_, data, "BoolValue");
                ^
/Users/coryan/cbt-develop/google/cloud/spanner/integration_tests/data_types_integration_test.cc:49:8: note: use non-reference type 'std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >'
  for (auto const& x : data) {
       ^~~~~~~~~~~~~~~
1 error generated.
```

--- a/google/cloud/spanner/integration_tests/data_types_integration_test.cc
+++ b/google/cloud/spanner/integration_tests/data_types_integration_test.cc
@@ -46,7 +46,7 @@ StatusOr<T> WriteReadData(Client& client, T const& data,
                           std::string const& column) {
   Mutations mutations;
   int id = 0;
-  for (auto const& x : data) {
+  for (auto&& x : data) {
     mutations.push_back(MakeInsertMutation("DataTypes", {"Id", column},
                                            "Id-" + std::to_string(id++), x));
   }

--- a/google/cloud/spanner/value.h
+++ b/google/cloud/spanner/value.h
@@ -420,7 +420,7 @@ class Value {
     *t.mutable_array_element_type() = MakeTypeProto(v.empty() ? T{} : v[0]);
     // Checks that vector elements have exactly the same proto Type, which
     // includes field names. This is documented UB.
-    for (auto const& e : v) {
+    for (auto&& e : v) {
       if (!google::protobuf::util::MessageDifferencer::Equals(
               MakeTypeProto(e), t.array_element_type())) {
         google::cloud::internal::ThrowInvalidArgument("Mismatched types");

